cmd/build/v2/smartbuild/hasher.go

```diff
@@ -149,3 +142,19 @@ func getDockerfileContent(dockerfileContext, dockerfilePath string) string {
 	encodedFile := sha256.Sum256(content)
 	return hex.EncodeToString(encodedFile[:])
 }
+
+func (sh *serviceHasher) getBuildContextHashInCache(buildContext string) string {
+	sh.lock.RLock()
```

Comment:
do we need a lock to read values?

Reply:
Yep, We are checking if the images are already built and given that several images access to the same map the lock is needed